misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/comparator/DumbNodeListComparator.java

```diff
@@ -69,20 +70,26 @@ public Optional<? extends NodeListDiffImpl<? extends Node>> computeDiff() {
         diffBuilder.withVersionImpact(SemverImpact.AMBIGUOUS);
 
         if (newNodesList != null && !newNodesList.isEmpty() && oldNodesList == null) {
-            String kind = nodeKindName != null ? nodeKindName : getNodeKindName(newNodesList.get(0).kind());
-            diffBuilder.withType(DiffType.NEW).withMessage(String.format("a new %s list is added", kind));
+            String kind = nodeKind != null ? nodeKind.toString() : getNodeKindName(newNodesList.get(0).kind());
```

Comment:
Why do we only concern the first element in the newNodesList without reset of the elements of nodes?

Reply:
Because here we need to derive the element type of the array. Hence using the 0th element to get the element type